src/3d/qgs3danimationsettings.h

```diff
@@ -63,6 +63,11 @@ class _3D_EXPORT Qgs3DAnimationSettings
     //! Returns the interpolation method for transitions of the camera
     QEasingCurve easingCurve() const { return mEasingCurve; }
 
+    //! Sets the animation widget visibility
+    void setWidgetVisible( const bool visible ) { mWidgetVisible = visible; }
```

Comment:
This is a bit of a bleed of gui into core 3d concepts. It'd be nicer if this was handled in the 3d map canvas/dock widget classes instead

Reply:
As this is only the setting class, LGTM

Reply:
@nyalldawson I'm not sure I understand your comment. `Qs3DAnimationSettings` is only consumed by `Qgs3DAnimationWidget` internally.

Could you please explain a little bit what would require your approach ?

Reply:
What I'm proposing is that instead of storing the visibility setting in Qs3DAnimationSettings we instead just directly write/read it to xml in eg qgisapp.cpp around lines 15955 / 15913 (ie at the same time as we use Qgs3dAnimationSettings writeXml/readXml

Reply:
Done.